src/OptionAndResult.dfy

```diff
@@ -0,0 +1,42 @@
+// RUN: %dafny "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+/*********************************************************************************************
+*  Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT 
+**********************************************************************************************/
+
+module OptionAndResult {
```

Comment:
For the record I don't think it would be terrible to have separate modules for this, if there's no logical grouping.

FYI I know of a Dafny project that uses the idiom of one definition per module for maximum flexibility, and to avoid repetitive names just calls the datatypes `Option.T` and `Result.T`, which I'm told is an established idiom in OCaml.

Reply:
Would we put `ToResult()` and `ToOption()` in a separate module to avoid a circular dependency?

Reply:
Shoot, good point. Okay, I can live with `OptionAndResult` unless anyone suggests better. :)